libs/common/graphics.zig

```diff
@@ -150,14 +140,57 @@ pub const GraphicsContext = struct {
             }
         }
 
+        const suitable_adapter = blk: {
+            var adapter: ?*dxgi.IAdapter1 = null;
+
+            var adapter_index: u32 = 0;
+            var optional_adapter1: ?*dxgi.IAdapter1 = undefined;
+
+            while (factory.EnumAdapterByGpuPreference(adapter_index, dxgi.DXGI_GPU_PREFERENCE_HIGH_PERFORMANCE, &dxgi.IID_IAdapter1, &optional_adapter1) == w.S_OK) {
+                if (optional_adapter1) |adapter1| {
+                    var adapter1_desc: dxgi.ADAPTER_DESC1 = undefined;
+                    if (adapter1.GetDesc1(&adapter1_desc) == w.S_OK) {
+                        if ((adapter1_desc.Flags & dxgi.DXGI_ADAPTER_FLAG_SOFTWARE) != 0) {
+                            // Don't select the Basic Render Driver adapter.
+                            continue;
+                        }
+
+                        var hr = d3d12.D3D12CreateDevice(@ptrCast(*w.IUnknown, adapter1), .FL_11_1, &d3d12.IID_IDevice9, null);
+                        // NOTE(gmodarelli): D3D12CreateDevice returns S_FALSE when the output device is null.
+                        // https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-d3d12createdevice#return-value
+                        if (hr == w.S_OK or hr == w.S_FALSE) {
+                            adapter = adapter1;
+                            break;
+                        }
+                    }
+                }
+
+                adapter_index += 1;
+            }
+
+            break :blk adapter;
+        };
+
         const device = blk: {
             var device: *d3d12.IDevice9 = undefined;
-            const hr = d3d12.D3D12CreateDevice(
-                null,
-                .FL_11_1,
-                &d3d12.IID_IDevice9,
-                @ptrCast(*?*c_void, &device),
-            );
+            const hr = innerblk: {
+                if (suitable_adapter) |adapter| {
```

Comment:
I think that we can simplify the code to this:
```
           const device = blk: {
                    var device: *d3d12.IDevice9 = undefined;
                    break :blk d3d12.D3D12CreateDevice(
                        if (suitable_adapter) |adapter| @ptrCast(*w.IUnknown, adapter) else null,
                        .FL_11_1,
                        &d3d12.IID_IDevice9,
                        @ptrCast(*?*c_void, &device),
                    );
            };
```

Reply:
Yes, that looks much better. Doing it now :)